src/Microsoft.Restier.Core/ApiBase.cs

```diff
@@ -137,6 +131,31 @@ public void Dispose()
         }
 
         /// <summary>
+        /// Configure the entire service set for this API. Descendants can override this method to choose
+        /// from the default service sets provided by RESTier.
+        /// </summary>
+        /// <param name="apiType">
+        /// CLR type of this <see cref="ApiBase"/>.
+        /// </param>
+        /// <param name="services">
+        /// The <see cref="IServiceCollection"/> with which to create an <see cref="ApiConfiguration"/>.
+        /// </param>
+        /// <returns>The <see cref="IServiceCollection"/>.</returns>
+        [CLSCompliant(false)]
+        protected virtual IServiceCollection ConfigureApiServices(Type apiType, IServiceCollection services)
+        {
+            var customizer = ApiConfiguration.Customize(apiType);
+            services.Apply(customizer.InnerMost);
+
+            return this.ConfigureApi(services)
+                .Apply(customizer.PrivateApi)
+                .UseAttributes(apiType)
+                .UseConventions(apiType)
+                .Apply(customizer.Overrides)
+                .Apply(customizer.OuterMost);
+        }
+
```

Comment:
1. I'm not suggesting abandoning ApiBase, it's very easy and intuitive for a quick start. I'm just asking if we want it to be the only option, to host an API. And I'm suggesting another option, which might be useful to advanced scenarios.
2. services is for transient usage only, need not to be a member.
3. Yeah I also think that's a problem, but forcing descendants to add all those default service sets is also painful.
4. IMHO with customizer presents, the need for overriding ApiBase should be minimized, common consumer could just register services with customizer anchors.